vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -11,14 +13,27 @@
 @Service
 public class ModuleManager implements IModuleManager {
 
-	@Autowired
-	private ModuleRepository moduleRepo;
-	
-	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.services.impl.IModuleManager#storeModule(edu.asu.diging.vspace.core.model.IModule, java.lang.String)
-	 */
-	@Override
-	public IModule storeModule(IModule module) {
-		return moduleRepo.save((Module)module);
-	}
+    @Autowired
+    private ModuleRepository moduleRepo;
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.diging.vspace.core.services.impl.IModuleManager#storeModule(edu.asu.
+     * diging.vspace.core.model.IModule, java.lang.String)
+     */
+    @Override
+    public IModule storeModule(IModule module) {
+        return moduleRepo.save((Module) module);
+    }
+
+    @Override
+    public IModule getModule(String id) {
+        boolean isExists = moduleRepo.existsById(id);
+        if (isExists) {
+            return moduleRepo.findById(id).get();
```

Comment:
nope, just do a 'isPresent' on the optional before calling get.

Reply:
moduleRepo.existById(id) is internally performing a isPresent() check in its method implementation. Also since I faced issue to test that condition in test cases, I changed it to existsById which can be mocked as it is a repo method.